retryablemessageprocessor.go

```diff
@@ -48,13 +48,13 @@ type RetryableMessageProcessor struct {
 // ProcessMessage invokes the wrapped `MessageProcessor`. Attempts retries using exponential backoff
 // if underlying 'MessageProcessor' returns an error.
 // If 'maxAttempts' are exceeded without successful processing, it emits a stat indicating the same
-func (t *RetryableMessageProcessor) ProcessMessage(ctx context.Context, record *kinesis.Record) MessageProcessError {
+func (t *RetryableMessageProcessor) ProcessMessage(ctx context.Context, record *kinesis.Record) MessageProcessorError {
 	stat := xstats.FromContext(ctx)
-	var messageProcErr MessageProcessError
+	var messageProcErr MessageProcessorError
 	var attemptNum int
 	for attemptNum < t.maxAttempts {
 		messageProcErr = t.wrapped.ProcessMessage(ctx, record)
-		if messageProcErr != nil && messageProcErr.Error() != nil && messageProcErr.IsRetryable() {
+		if messageProcErr != nil && messageProcErr.Error() != "" && messageProcErr.IsRetryable() {
```

Comment:
question: why have the check for the error string? Isn't the check for the error being non-nil and retryable sufficient to trigger the retry code path?

Reply:
Oops - my bad. this was according to original interface. Addressed this.